### PR TITLE
Create and document an MVP DB schema using DBML

### LINKED ISSUE #8: Schema setup instructions in README.md are incorrect
The instructions in README.md for setting up a new user and granting schema permissions for that new user `refactor_rs` don't give the right permissions needed to successfully perform the up migration.

The failed run looks like:

```sh
DATABASE_URL=postgres://refactor_rs:password@localhost:5432/refactor_platform_rs sea-orm-cli migrate up -s refactor_platform_rs
Running `cargo run --manifest-path ./migration/Cargo.toml -- up -u postgres://refactor_rs:password@localhost:5432/refactor_platform_rs -s refactor_platform_rs`
    Finished dev [unoptimized + debuginfo] target(s) in 0.21s
     Running `target/debug/migration up -u 'postgres://refactor_rs:password@localhost:5432/refactor_platform_rs' -s refactor_platform_rs`
Execution Error: error returned from database: no schema has been selected to create in
Fail to run migration
```

### PR DESCRIPTION
**Pull request by jhodapp**
## Description
This PR creates and documents an MVP DB schema for refactor_platform_rs using [DBML](https://dbml.dbdiagram.io/home/).

### Changes
* Creates/documents an MVP DB schema for refactor_platform_rs using DBML
* Adds a script to automatically generate Postgresql-compatible SQL and generate a new DB within a running instance of Postgres
* Updates the `users` entity with new fields
* Uses autoincrement integer-based internal primary keys and external UUID-based ids

DB schema diagram generated using dbdiagram.io's export feature:

Refactor Coaching Platform Schema v1

### Testing Strategy
1. Drop it in a new diagram session at [dbdiagram.io](https://dbdiagram.io/) ensuring there are no syntactical errors
2. Run `scripts/rebuild_db.sh` and make sure that it creates a new refactor_platform DB in Postgres
3. Ensure you're able to log in using the refactor-platform-fe login page

### Concerns
None

### CODE REVIEW

File: docs/db/refactor_platform_rs.dbml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  completed_at timestamp [note: 'The date and time an overarching goal was completed']
+  completed_by_id integer [note: 'The user that successfully completed an overarching goal']
+  created_at timestamp
+  updated_at timestamp [note: 'The last date and time an overarching goal\'s fields were changed']
+}
+
+Table refactor_platform.notes {
+  id integer [primary key, unique, not null, increment]
+  coaching_session_id integer
+  body varchar [note: 'Main text of the note supporting Markdown']
```
Comment (calebbourg) [inline docs/db/refactor_platform_rs.dbml:77]: This and some of the other larger text fields (Ex `agreements.details` below) might be good candidates for the `text` type.

File: docs/db/refactor_platform_rs.dbml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+Table refactor_platform.coaching_sessions {
+  id integer [primary key, unique, not null, increment]
+  coaches_coachees_id integer [note: 'The coaching relationship (i.e. what coach & coachee under what organization) associated with this coaching session']
+  date timestamp [note: 'The date and time of a session']
+  timezone timezone [note: 'The baseline timezone used for the `date` field']
+  created_at timestamp
+}
+
+Table refactor_platform.overarching_goals {
```
Comment (calebbourg) [inline docs/db/refactor_platform_rs.dbml:62]: Love this

File: docs/db/refactor_platform_rs.dbml [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  title varchar [note: 'A short description of an overarching goal']
+  details varchar [note: 'A long description of an overarching goal']
+  completed boolean // May be unnecessary if there's a valid completed_at timestamp
+  completed_at timestamp [note: 'The date and time an overarching goal was completed']
+  completed_by_id integer [note: 'The user that successfully completed an overarching goal']
+  created_at timestamp
+  updated_at timestamp [note: 'The last date and time an overarching goal\'s fields were changed']
+}
+
+Table refactor_platform.notes {
```
Comment (calebbourg) [inline docs/db/refactor_platform_rs.dbml:74]: For some of these resources. Would it be valuable to add a `user_id` which would tie the record to "the user who created the resource"?

That might only be applicable for notes.

File: docs/db/refactor_platform_rs.dbml (hunk 2 above)
Comment (jhodapp) [reply docs/db/refactor_platform_rs.dbml:62]: Thanks, glad to hear that!

File: docs/db/refactor_platform_rs.dbml (hunk 3 above)
Comment (jhodapp) [reply docs/db/refactor_platform_rs.dbml:74]: @calebbourg 100%, I thought about who this was associated with coachee/coach-wise, but forgot what you suggested - the creator of the note. Great suggestion.

File: docs/db/refactor_platform_rs.dbml (hunk 1 above)
Comment (jhodapp) [reply docs/db/refactor_platform_rs.dbml:77]: @calebbourg great suggestion, I agree.

**Review by calebbourg — APPROVED** [review]

**Pull request merged**